pkg/pub_worker/lib/src/testing/server.dart

```diff
@@ -0,0 +1,296 @@
+/// A test server for pub_worker
+library server;
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:_pub_shared/data/package_api.dart';
+import 'package:_pub_shared/data/task_api.dart';
+import 'package:async/async.dart';
+import 'package:collection/collection.dart';
+import 'package:http_parser/http_parser.dart';
+import 'package:indexed_blob/indexed_blob.dart';
+import 'package:mime/mime.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:pubspec_parse/pubspec_parse.dart';
+import 'package:shelf/shelf.dart';
+import 'package:shelf/shelf_io.dart' as io;
+import 'package:shelf_router/shelf_router.dart';
+import 'package:tar/tar.dart';
+import 'package:yaml/yaml.dart';
+import 'package:http/http.dart' as http;
+import 'package:retry/retry.dart';
+
+part 'server.g.dart';
+
+class PubWorkerTestServer {
```

Comment:
Would this be served better with running `../../app/bin/fake_server.dart` instead? Is there something limiting there?